beetsplug/fetchart.py

```diff
@@ -329,20 +329,36 @@ def get(self, album, plugin, paths):
         if plugin.maxwidth in self.VALID_THUMBNAIL_SIZES:
             size_suffix = "-" + str(plugin.maxwidth)
 
+        url = None
         if 'release' in self.match_by and album.mb_albumid:
-            if size_suffix:
-                release_thumbnail_url = release_url + size_suffix
-                yield self._candidate(url=release_thumbnail_url,
+            url = release_url
+        elif 'releasegroup' in self.match_by and album.mb_releasegroupid:
+            url = release_group_url
```

Comment:
Previously, the art source would yield candidates from both the release and the release group, with match types `MATCH_EXACT` and `MATCH_FALLBACK`, respectively. Now, if for example, the album has both `release` and `releasegroup` set, and there's no art for the release, but for the group there is, the art source will not return it. Maybe a loop like `for match_by in ("release", "releasegroup")` could restore the behaviour?

Reply:
Good catch. Fixed in the latest commit.